src/consolidator/schema_merger.py

```diff
@@ -0,0 +1,136 @@
+        output_dir (str): Directory to write the merged schema files.
+    """
+    os.makedirs(output_dir, exist_ok=True)
+
+    for schema_file in schema_files:
+        name = schema_file.split(".")[0]
+        base_path = os.path.join(input_dir, schema_file)
+
+        try:
+            with open(base_path, 'r', encoding='utf-8') as f:
+                data = f.read()
+
+            
+
+            for dep in combined_deps.get(name, []):
+                dep_path = os.path.join(input_dir, f"{dep}.asn1")
+                with open(dep_path, 'r', encoding='utf-8') as dep_file:
+                    data += dep_file.read()
+
+            out_path = os.path.join(output_dir, schema_file)
+            with open(out_path, 'w', encoding='utf-8') as out_file:
+                out_file.write(data)
+
+        except Exception as e:
+            logger.error(f"Failed to merge schema: {schema_file}. Reason: {e}")
+
+def wrap_definitions(schema_files, target_dir="data/files/schema_updated"):
+    """
+    Wraps each ASN.1 file in `DEFINITIONS ::= BEGIN ... END` block.
+
+    Args:
+        schema_files (list[str]): List of schema filenames.
+        target_dir (str): Directory containing updated schema files.
+    """
+    for schema_file in schema_files:
+        file_path = os.path.join(target_dir, schema_file)
+        name = schema_file.split(".")[0]
+
+        try:
+            with open(file_path, 'r', encoding='utf-8') as file:
+                lines = file.read().splitlines()
+
+           
+
+            # Prepend and append definition markers
+            wrapped_lines = [f"{name} DEFINITIONS ::= BEGIN"] + lines + ["END"]
+
+            with open(file_path, 'w', encoding='utf-8') as file:
+                for line in wrapped_lines:
+                    file.write(line + '\n')
+
+        except Exception as e:
+            logger.error(f"Failed to wrap ASN.1 definitions in {schema_file}: {e}")
+
+def verify_syntax(schema_files, target_dir="data/files/schema_updated", error_log_path="error_files.json"):
+    """
+    Verifies ASN.1 syntax and structural correctness using asn1tools' internal parser.
+
+    This does NOT compile to encoding/decoding rules but ensures the file is well-formed.
+
+    Args:
+        schema_files (list[str]): List of schema filenames to check.
+        target_dir (str): Directory where schema files are located.
+        error_log_path (str): Path to output JSON file containing syntax errors.
+    """
+    error_count = 0
+    error_files = {}
+
+    for schema_file in schema_files:
+        file_path = os.path.join(target_dir, schema_file)
+      
+
+        try:
+            # Use parse_files to parse the syntax and validate syntax
+            asn1tools.parse_files(file_path)
+
+        except Exception as e:
+          
+            error_files[schema_file] = str(e)
+            error_count += 1
+
+   
+
+    if error_files:
+        with open(error_log_path, 'w', encoding='utf-8') as file:
```

Comment:
Would it be preferable to output to stderr instead of a file? Might be easier to integrate with other tools in a CI pipeline